### PR TITLE
Adding PyTorch + DNNL + AMD BLIS path

### PR DESCRIPTION
**Pull request by shrutiramesh1988**
These changes provide user with an additional option to choose DNNL+BLIS path for PyTorch.
This assumes BLIS is already downloaded or built from source and the necessary library file is available at the location: $BLIS_HOME/lib/libblis.so and include files are available at: $BLIS_HOME/include/blis/blis.h and $BLIS_HOME/include/blis/cblas.h

Export the below variables to build PyTorch with MKLDNN+BLIS and proceed with the regular installation procedure as below:
$export BLIS_HOME=path-to-BLIS
$export PATH=$BLIS_HOME/include/blis:$PATH LD_LIBRARY_PATH=$BLIS_HOME/lib:$LD_LIBRARY_PATH
$export BLAS=blis USE_MKLDNN_CBLAS=ON WITH_BLAS=blis
$python setup.py install

CPU only Dockerfile is also available at : docker/pytorch/cpu-only/Dockerfile

Fixes #{issue number}

### CODE REVIEW

**Review by robieta — APPROVED** [review]
LGTM. Adding Nikita to make sure everything is in order.

**Review by malfet — CHANGES_REQUESTED** [review]
Thank you very much for the change. Overall looks good to me, but I have a question regarding filenames:
- Shouldn't BLIS be capitalized, i.e. FindBLIS.cmake, rather than `Findblis.cmake`?
- Can you please rename `cpu-only/Dockerfile` into `cpu-bliss/Dockerfile` or `blis/Dockerfile`
- Looks like BLIS library is Linux only (isn't it?) In that case, can you please guard checks for BLIS with appropriate OS guards, or at least add it to the documentation.
Nits: can we just sort available blas libraries options alphabetically?  (will add suggestions to PR)

File: cmake/Dependencies.cmake [hunk 1]
```diff
@@ -118,7 +118,7 @@ else()
   set(AT_MKLDNN_ENABLED 0)
   set(AT_MKL_ENABLED 0)
 endif()
-set_property(CACHE BLAS PROPERTY STRINGS "Eigen;ATLAS;OpenBLAS;MKL;vecLib;FLAME;Generic")
+set_property(CACHE BLAS PROPERTY STRINGS "Eigen;ATLAS;OpenBLAS;blis;MKL;vecLib;FLAME;Generic")
```
Comment (malfet) [inline cmake/Dependencies.cmake:121]: Is order important here? If not, can we just sort the alphabetically?
```suggestion
set_property(CACHE BLAS PROPERTY STRINGS "ATLAS;Eigen;FLAME;Generic;MKL;OpenBLAS;blis;vecLib")
```

File: cmake/Modules/FindBLAS.cmake [hunk 2]
```diff
@@ -20,7 +20,7 @@ SET(BLAS_INCLUDE_DIR)
 SET(BLAS_INFO)
 SET(BLAS_F2C)
 
-SET(WITH_BLAS "" CACHE STRING "Blas type [mkl/open/goto/acml/atlas/accelerate/veclib/generic]")
+SET(WITH_BLAS "" CACHE STRING "Blas type [blis/mkl/open/goto/acml/atlas/accelerate/veclib/generic]")
```
Comment (malfet) [inline cmake/Modules/FindBLAS.cmake:23]: Can we sort library names alphabetically?
Also, shouldn't their capitalization match that of `BLAS` property? 
```suggestion
SET(WITH_BLAS "" CACHE STRING "Blas type [accelerate/acml/atlas/blis/generic/goto/mkl/open/veclib]")
```

File: cmake/Dependencies.cmake (hunk 1 above)
Comment (shrutiramesh1988) [reply cmake/Dependencies.cmake]: Thanks for the suggestions, I've addressed these in the latest commit. Order is not important, sorted them alphabetically. Also changed blis to BLIS

File: cmake/Modules/FindBLAS.cmake (hunk 2 above)
Comment (shrutiramesh1988) [reply cmake/Modules/FindBLAS.cmake]: Sorted them as suggested, however retained blis in small letters to match that of mkl's convention. BLAS is BLIS/MKL and WITH_BLAS is blis/mkl

**Comment by shrutiramesh1988** [comment]
Will raise a new PR with suggested changes as something went wrong with the latest commit.

**Comment by shrutiramesh1988** [comment]
Kindly review the updated PR : https://github.com/pytorch/pytorch/pull/53985

**Pull request closed without merging**